mantle/kola/harness.go

```diff
@@ -242,9 +243,21 @@ func filterTests(tests map[string]*register.Test, patterns []string, pltfrm stri
 		checkPlatforms = append(checkPlatforms, "qemu")
 	}
 
+	var noNetFiltered bool
 	var blacklisted bool
 	noPattern := hasString("*", patterns)
 	for name, t := range tests {
+		for _, flag := range t.Flags {
```

Comment:
Hmm, shouldn't we reset `noNetFiltered` to `false` before this? Otherwise once it's set to `true`, it'll always remain `true`, no? Which means it would skip every test after a networking test when `--no-net` is specified.

Or better yet, just move the variable declaration into the for-loop. I guess we could do the same for `blacklisted` too.

Reply:
Oooh great catch!  I'd tested that we didn't run the network tests *and* that other tests were run, but not that we still listed *all* of the tests...

Reply:
Happened to still have this tab open and one thing I wanted to say here is that this is an excellent example of why we have code review - nothing would be checking today that we're silently not running some tests and it'd be painful to find later.